scripts/benchmark-rule.mjs

```diff
@@ -64,8 +64,19 @@ const [primary, secondary] = ruleSettings;
 const context = ruleContext ? JSON.parse(ruleContext) : {};
 
 const rule = (await rules[ruleName])(primary, secondary, context);
-
-const processor = postcss().use(rule);
+const fn = (root, result) => {
+	result.stylelint ||= {
+		ruleSeverities: {},
+		customMessages: {},
+		customUrls: {},
+		fixersData: {},
+		ruleMetadata: {},
+		disabledRanges: {},
+	};
```

Comment:
@ybiquitous I chose not to revert your changes to the `report` utility.
i.e. it could also be done directly in `report` but that couldn't be justified by the current type of `PostcssResult['stylelint']`

Reply:
I think we should modify this benchmark script by using the public API, such as `utils.checkAgainstRule`, instead of increasing copy-and-paste.

Reply:
I just tried another approach with PR #8090, but this is quite slow (about 2x slower in my machine).